test/pipelinefinally_test.go

```diff
@@ -201,13 +247,16 @@ func TestPipelineLevelFinally_OneDAGTaskFailed_InvalidTaskResult_Failure(t *test
 		t.Fatalf("Error listing TaskRuns for PipelineRun %s: %s", pipelineRun.Name, err)
 	}
 
-	// expecting taskRuns for dagtask1, dagtask2, dagtask3 (with condition failure), dagtask5, finaltask1, finaltask2, and finaltaskconsumingdagtask5
-	if len(taskrunList.Items) != 7 {
+	// expecting taskRuns for dagtask1, dagtask2, dagtask3 (with condition failure), dagtask5, finaltask1, finaltask2,
+	// finaltaskconsumingdagtask5, finaltaskconsumingdagtask5inwhenexpressionstrue, finaltaskconsumingdagtask5executionstatusinwhenexpressionstrue
```

Comment:
i feel like these names are getting a bit long - i wonder if there is some other way we can make this clearer 🤔

Reply:
shortened them and added comments to describe them